lib/Map/Reproject.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+/*global require*/
+var proj4 = require('proj4/lib/index.js');
+
+var loadText = require('terriajs-cesium/Source/Core/loadText');
+var defined = require('terriajs-cesium/Source/Core/defined');
+var Proj4Definitions = require('../Map/Proj4Definitions');
+var urijs = require('urijs');
+var when = require('terriajs-cesium/Source/ThirdParty/when');
+
+
+var Reproject = {
+    TERRIA_CRS: 'EPSG:4326',
+
+    /**
+     * Convert a CRS string into a code that Proj4 understands.
+     * @param {String} crsString A CRS URI
+     * @return {String} A code that is recognised by Proj4 as a valid CRS code.
+     */
+    crsStringToCode: function(crsString) {
+        var crs = crsString.toLowerCase();
+        var code;
+        if (crs.indexOf('epsg:') === 0) {
+            code = crsString.toUpperCase();
+        } else if (crs.indexOf('urn:ogc:def:crs:epsg::') === 0) {
+            code = 'EPSG:' + crsString.substring('urn:ogc:def:crs:EPSG::'.length);
+        } else if (crs.indexOf('urn:ogc:def:crs:epsg') === 0) {
+            // With version
+            code = 'EPSG:' + crsString.substring('urn:ogc:def:crs:EPSG::x.x'.length);
+        } else if (crs.indexOf('crs84') !== -1) {
+            code = this.TERRIA_CRS;
```

Comment:
CRS84 and EPSG:4326 have opposite coordinate order, right?  I don't see that being handled anywhere.